src/ui/MessageInput/index.tsx

```diff
@@ -28,20 +28,29 @@ import { OpenChannel } from '@sendbird/chat/openChannel';
 import { UserMessage } from '@sendbird/chat/message';
 
 const TEXT_FIELD_ID = 'sendbird-message-input-text-field';
-const LINE_HEIGHT = 76;
-const DEFAULT_CHAT_VIEW_HEIGHT = 600;
 const noop = () => {
   return null;
 };
 
-const displayCaret = (element: HTMLInputElement, position: number) => {
-  const range = document.createRange();
-  const sel = window.getSelection();
-  range.setStart(element.childNodes[0], position);
-  range.collapse(true);
-  sel?.removeAllRanges();
-  sel?.addRange(range);
-  element.focus();
+const scrollToCaret = () => {
+  const selection = window.getSelection();
+  if (selection && selection.rangeCount > 0) {
+    const range = selection.getRangeAt(0);
+    const caretNode = range.endContainer;
+
+    // Ensure the caret is in a text node
+    if (caretNode.nodeType === NodeTypes.TextNode) {
+      const parentElement = caretNode.parentElement;
+
+      if (parentElement && parentElement.scrollIntoView) {
+        // Scroll the parent element of the caret into view
+        parentElement.scrollIntoView({
+          behavior: 'smooth',
+          block: 'nearest',
+        });
+      }
```

Comment:
```suggestion
      // Scroll the parent element of the caret into view
      parentElement?.scrollIntoView?.({
        behavior: 'smooth',
        block: 'nearest',
      });
```
nit; we can remove one more if stmt considering the newly added block has 3 if statements inside.